Source/sound.cpp

```diff
@@ -261,82 +260,69 @@ TSnd *__fastcall sound_file_load(char *path)
 
 void __fastcall sound_CreateSoundBuffer(TSnd *sound_file)
 {
-	TSnd *v1; // esi
-	int v2; // eax
-	DSBUFFERDESC v3; // [esp+4h] [ebp-14h]
-
-	v1 = sound_file;
-	memset(&v3, 0, sizeof(DSBUFFERDESC));
-	v3.dwBufferBytes = v1->len;
-	v3.lpwfxFormat = &v1->fmt;
-	v3.dwSize = sizeof(DSBUFFERDESC);
-	v3.dwFlags = DSBCAPS_CTRLVOLUME|DSBCAPS_CTRLPAN|DSBCAPS_STATIC;
-	v2 = sglpDS->CreateSoundBuffer(&v3, &v1->DSB, NULL);
-	if ( v2 )
-		DSErrMsg(v2, 282, "C:\\Src\\Diablo\\Source\\SOUND.CPP");
+	DSBUFFERDESC DSB;
+	memset(&DSB, 0, sizeof(DSBUFFERDESC));
+	DSB.dwBufferBytes = sound_file->len;
+	DSB.lpwfxFormat = &sound_file->fmt;
+	DSB.dwSize = sizeof(DSBUFFERDESC);
+	DSB.dwFlags = DSBCAPS_CTRLVOLUME|DSBCAPS_CTRLPAN|DSBCAPS_STATIC;
+	int soundBuffer = sglpDS->CreateSoundBuffer(&DSB, &sound_file->DSB, NULL);
+
+	if ( soundBuffer )

```

Comment:
"int soundbuffer" should probably be "HRESULT hr" and this comparison "if ( hr != ERROR_SUCCESS )"